mslice/models/workspacemanager/workspace_algorithms.py

```diff
@@ -106,35 +104,34 @@ def _get_property_from_history(name, history):
     return None
 
 
-def get_q_limits(theta, emax, efix):
-    qmin, qmax, qstep = tuple(np.sqrt(E2q * 2 * efix * (1 - np.cos(theta)) * meV2J) / m2A)
-    qmax = np.sqrt(E2q * (2 * efix + emax - 2 * np.sqrt(efix * (efix + emax)) * np.cos(theta[1])) * meV2J) / m2A
+def get_q_limits(theta, en, efix, ndetectors):
+    #calculates the Q(E) line for the given two theta and then finds the min and max values
+    qlines = [np.sqrt(E2q * (2 * efix - en - 2 * np.sqrt(efix * (efix - en)) * np.cos(tth)) * meV2J) / 1e10 for tth in theta[:2]]
+    qmin = np.nanmin(qlines[0])
+    qmax = np.nanmax(qlines[1])
+
+    if(ndetectors > 1000):
+        qstep = (qmax - qmin) / 100
+    else:
+        qstep = np.sqrt(E2q * 2 * efix * (1 - np.cos(theta[2])) * meV2J) / m2A
+        qstep = qstep / 3  # Use a step size a bit smaller than angular spacing ( / 3) so user can rebin if they want
```

Comment:
Also, we can remove this line I think. I don't know what I was thinking when I put this in... (I think it might have been when we were using this as the minimum and users could not put in smaller steps, in case we run out of memory for the slice).